R/add_weights_cluster.R

```diff
@@ -119,19 +119,86 @@ add_weights_cluster <- function(x, cl,
                                 cluster_x = NULL, cluster_cl = NULL,
                                 household_x = NULL, household_cl = NULL,
                                 ignore_cluster = TRUE, ignore_household = TRUE,
-                                surv_weight = "surv_weight", 
+                                surv_weight = "surv_weight",
                                 surv_weight_ID =  "surv_weight_ID") {
 
- 
+
 
   # define vars
-  clus_id_cl <- tidyselect::vars_select(colnames(cl), {{cluster_cl}})
-  hh_id_cl   <- tidyselect::vars_select(colnames(cl), {{household_cl}})
-  clus_id_x  <- tidyselect::vars_select(colnames(x),  {{cluster_x}})
-  hh_id_x    <- tidyselect::vars_select(colnames(x),  {{household_x}})
-  indiv_eli  <- tidyselect::vars_select(colnames(x),  {{eligible}})
-  indiv_surv <- tidyselect::vars_select(colnames(x),  {{interviewed}})
+  clus_id_cl <- tidyselect::vars_select(colnames(cl), {{cluster_cl}}, .strict = FALSE)
+  hh_id_cl   <- tidyselect::vars_select(colnames(cl), {{household_cl}}, .strict = FALSE)
+  clus_id_x  <- tidyselect::vars_select(colnames(x),  {{cluster_x}}, .strict = FALSE)
+  hh_id_x    <- tidyselect::vars_select(colnames(x),  {{household_x}}, .strict = FALSE)
+  indiv_eli  <- tidyselect::vars_select(colnames(x),  {{eligible}}, .strict = FALSE)
+  indiv_surv <- tidyselect::vars_select(colnames(x),  {{interviewed}}, .strict = FALSE)
+
+  ## checks (errors / warnings)
+
+  # check if variables are in datasets (spelling) - throws error
+
+  # for cluster dataset
+  if (length(clus_id_cl) == 0) {
+    # pull the environment from which this was called (i.e. the function)
+    cll   <- match.call()
```

Comment:
I would suggest to move the `match.call()` to the top of the function. You can refer to it in each of these errors later. That being said, it's not a deal breaker.

Reply:
yes please i just couldnt check if it worked when i was writing it - but if it works then cool

Reply:
fixed in latest commit